runtime/flang/miscsup_com.c

```diff
@@ -4563,13 +4568,16 @@ ENTF90(NEARESTDX, nearestdx)(__REAL8_T d, __LOG_T sign)
   if (x.d == 0.0) {
     x.i.h = (sign & 1) ? 0x00100000 : 0x80100000;
     x.i.l = 0;
-  } else {
-    if ((x.ll >> 52 & 0x7FF) != 0x7FF) { /* not nan or inf */
+  } else if ((x.ll >> 52 & 0x7FF) != 0x7FF) { /* not nan or inf */
       if ((x.d < 0) ^ (sign & GET_DIST_MASK_LOG))
         ++x.ll;
       else
         --x.ll;
-    }
+  } else {
+    if (~sign && (x.ll & ~0x0) == 0x7FF0000000000000)
+      x.ll = 0x7FEFFFFFFFFFFFFF;
```

Comment:
Here as well, consider breaking this `else` block into two `else if` blocks.

Reply:
I've modified the code as suggested. Thanks!

Reply:
I've modified the code as suggested. Thanks!